### DISCUSSION TITLE
Bug With Anonymous Records?

### DISCUSSION THREAD

**Discussion by cwarden** [souffle-lang/souffle#2156] [Q&A] [answered]
I'm working with records for the first time and running into an unexpected issue.  Running the following program with `souffle -v`, souffle seems to get stuck in a loop of `ResolveAnonymousRecordAliases` and `FoldAnonymousRecords`.

```
.type Cell = [
   row: number,
   column: number
]

.decl cell_size(cell: Cell, size: number)
.decl number_bigger(cell: Cell, num: number)

cell_size([0, 1], 5).
cell_size([1, 1], 3).

number_bigger([r1, p1], n) :-
   cell_size([r1, p1], A),
   n = count : { cell_size([r2, p2], B),
   [r1, p1] != [r2, p2],
   A < B}.

.output number_bigger(IO=stdout, headers=true)
```

Is it a bug or am I doing something wrong?

**Comment by b-scholz** [comment, marked as answer]
This seems to be a bug - please, can you post it on the issue list. Is it the latest version of Souffle?
You can rewrite the code above as follows:
```
.type Cell = [
   row: number,
   column: number
]

.decl cell_size(cell: Cell, size: number)
.decl number_bigger(cell: Cell, num: number)

cell_size([0, 1], 5).
cell_size([1, 1], 3).

number_bigger([r1, p1], n) :-
   cell_size([r1, p1], A),
   n = count : { cell_size([r2, p2], B),
   r1 != r2,
   p1 != p2,
   A < B}.

.output number_bigger(IO=stdout, headers=true)
```
and you may be luckier but I see that the aggregate transformer is failing as well.

**Reply by cwarden** [reply]
Thanks.  I created #2158 and noted that it occurs as of 244cb0718d1c0aa6e2bd717fb316c4680c7a8965.
